src/games/strategy/triplea/ai/proAI/util/ProMoveUtils.java

```diff
@@ -194,7 +194,8 @@ public static void calculateAmphibRoutes(final PlayerID player, final List<Colle
                   && maxUnitDistance <= minUnitDistance
                   && distanceFromUnloadTerritory < movesLeft
                   && (maxUnitDistance < minUnitDistance
-                      || (maxUnitDistance > 1 && neighborDistanceFromEnd > maxDistanceFromEnd) || (maxUnitDistance <= 1 && neighborDistanceFromEnd < maxDistanceFromEnd))) {
+                      || (maxUnitDistance > 1 && neighborDistanceFromEnd > maxDistanceFromEnd)
```

Comment:
This if statement is complex. This will be hard to maintain, and complex boolean conditions have a higher correlation with defective code. Is there anything that can be done? 
The series of 'if' statements suggest a decomposition into polymorphic objects refactor could simplify this.